crates/ruma-common/src/events/room/redaction.rs

```diff
@@ -284,18 +354,46 @@ impl From<RoomRedactionEvent> for SyncRoomRedactionEvent {
     }
 }
 
-impl<'de> Deserialize<'de> for SyncRoomRedactionEvent {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let json = Box::<RawJsonValue>::deserialize(deserializer)?;
-        let RedactionDeHelper { unsigned } = from_raw_json_value(&json)?;
-
-        if unsigned.and_then(|u| u.redacted_because).is_some() {
-            Ok(Self::Redacted(from_raw_json_value(&json)?))
-        } else {
-            Ok(Self::Original(from_raw_json_value(&json)?))
+impl OriginalRoomRedactionEvent {
+    /// Returns the ID of the event that this event redacts, according to the given room version.
+    ///
+    /// Returns `None` if the only available `redacts` field is not the proper one for the given
+    /// room version.
+    pub fn redacts(&self, room_version: &RoomVersionId) -> Option<&EventId> {
```

Comment:
To be more useful we probably should always return something, even if it's the wrong field, since we are sure from deserialization that at least one of the two fields is set.

Reply:
Okay, but let's log an error when things are not as expected.